src/wallet/walletdb.cpp

```diff
@@ -522,13 +522,13 @@ bool CWalletDB::IsKeyType(const std::string& strType)
             strType == "mkey" || strType == "ckey");
 }
 
-DBErrors CWalletDB::LoadWallet(CWallet* pwallet)
+DBErrors CWalletDB::LoadWallet(CWallet* pwallet) EXCLUSIVE_LOCKS_REQUIRED(pwallet->cs_wallet)
```

Comment:
Adding this only in the .cpp means that the lock requirement will only be enforced for other functions in this file.

I realize that it can't be added in the header as-is due to the circular dependency, but that needs to be fixed somehow.

Reply:
@theuni Yes, I'd like to put the annotation in the header file but I need help with resolving the circular dependency. What would be the cleanest way to solve it? :-)

Reply:
I'm afraid I don't see a simple solution :(

Reply:
I don't know if it's a simple solution, but it looks like you can resolve the circular dependency by declaring:

    inline CCriticalSection &CWallet_cs_wallet(CWallet *pwallet) LOCK_RETURNED(pwallet->cs_wallet)
    {
        return pwallet->cs_wallet;
    }

in wallet.h and

    CCriticalSection &CWallet_cs_wallet(CWallet *pwallet);
    class CWalletDB {
        DBErrors LoadWallet(CWallet* pwallet) EXCLUSIVE_LOCKS_REQUIRED(CWallet_cs_wallet(pwallet));
    }

in walletdb.h.